Add test code for archive feature (#115)
AY2223S1-CS2103-F14-3/tp: docs/UserGuide.md

   
    * **`edit`**`1 e/hr@internship.sg`: Edits the 1st application shown in the current list with new email address.
   

>     * **`archive`**`3`: Archives the 3rd application shown in the current list. The application must be not archived.

Reviewer: ```suggestion
    * **`archive`**`3`: Archives the 3rd application shown in the current list. The application must be unarchived.
```

`edit 1 ct/91234567 e/grab@example.com` edits the contact and email of the 1st application to be 91234567 and grab@example.com respectively.
`edit 2 c/Garena t/` edits the company of the 2nd application to be `Garena` and clears all existing tags.

### Archiving an internship application: `archive`

Archives the specified internship application from the list.

Format: `archive INDEX`

* Archives the internship application at the specified `INDEX`.

> * The application specified must be not archived.

Reviewer: ```suggestion
* The application specified must be unarchived.
```

* The index refers to the index number shown in the displayed internship application list.
* The index **must be a positive integer** 1, 2, 3, …​
* The index must be within the range of available internships in the shown application list.

> Examples:

Reviewer: ```suggestion
Example:
```

* `list` followed by `archive 2` archives the 2nd internship application in the list.

### Retrieving an internship application from archives: `retrieve`

Retrieves the specified internship application from the list.

Format: `retrieve INDEX`

* Retrieves the internship application at the specified `INDEX`.

> * The application specified must be archived, `list-archive` command is recommended to be used before using this command.

Reviewer 2: is it just recommended to use `list-archive` or is it compulsory?

Author: recommended, just that if retrieving one application that has not been archived will throw command exception. The exception will also tell user to use `list-archive`. 

Considering user will also have chance in to use archive after find command, so it doesn't mean that it will always throw command exception if `list-archive` is not used.

Reviewer 2: ah okay so there has been a change in behaviour since the last implementation? I tried running my version of it and using `retrieve` without using `list-archive` and it shows an error 

![image](https://user-images.githubusercontent.com/26516640/197820788-dd1c2cbb-d073-4723-bdd1-76634b159e69.png)

Edit: I get what you mean now by first using `find` then `retrieve`. But if users are on the main list itself they cannot retrieve, is that right?

Author: Yup, because you are retrieving the application that is not archived. 

So usually `list-archive` should be use in order to ensure that the application shown in the UI are all archived, but considering if user using find command which will also possible to search for archived application and show on the UI, I put "recommended" instead of "must" in the user guide.

Reviewer 2: okay got it! 👍

* The index refers to the index number shown in the displayed internship application list.
* The index **must be a positive integer** 1, 2, 3, …​
* The index must be within the range of available internships in the shown application list.

> Examples:

Reviewer: ```suggestion
Example:
```

> * `list-archive` followed by `retrieve 1` retrieves the 2nd internship application in the list.

Reviewer 2: small typo

```suggestion
* `list-archive` followed by `retrieve 1` retrieves the 1st internship application in the list.
```
